projects/saturn/src/main/java/io/fairspace/saturn/services/search/AiSearchApp.java

```diff
@@ -0,0 +1,222 @@
+                }
+
+                var query = body.getString("querytext");
+                var conversationId = body.getString("conversationId");
+
+                var result = new LlmConversation().continueChat(conversationId, query);
+                var filepath = CACHE_DIR + getUserKey();
+                var file = new java.io.File(filepath + "/" + conversationId + ".json");
+
+                Files.createDirectories(Paths.get(filepath));
+
+                try (FileWriter writer = new FileWriter(file)) {
+                    writer.write(result.toString());
+                } catch (Exception e) {
+                    System.out.println("Error writing conversation history file: " + e.getMessage());
+                }
+
+                res.type(APPLICATION_JSON.asString());
+                return result;
+            } catch (Exception e) {
+                return handleError(req, e);
+            }
+        });
+
+        post("/deletechat/:id",  "application/json", (req, res) -> {
+            try {
+                boolean keyExists = req.params() != null && req.params().containsKey(":id");
+
+                if (!keyExists) {
+                    return "Please provide a valid conversationid.";
+                }
+
+                var conversationId = req.params(":id");
+                var filename = CACHE_DIR + getUserKey() + "/" + conversationId + ".json";
+
+                try {
+                    Files.deleteIfExists(Paths.get(filename));
+                } catch (Exception e) {
+                    System.out.println("Error deleting conversation history file: " + e.getMessage());
+                }
+
+                var result = new LlmConversation().deleteChat(conversationId);
+
+                res.type(APPLICATION_JSON.asString());
+                return result;
+            } catch (Exception e) {
+                System.out.println("Error deleting conversation history file: " + e.getMessage());
+                return handleError(req, e);
+            }
+        });
+
+        post("/", "application/json", (req, res) -> {
+            try {
+                var body = new JSONObject(req.body());
+
+                boolean keyExists = body.has("querytext");
+
+                if (!keyExists) {
+                    return "Please provide a querytext key in the request body";
+                }
+
+                var query = body.getString("querytext");
+                var searchResult = new LlmConversation().startChatWithQuery(query, getUserKey());
+                res.type(APPLICATION_JSON.asString());
+
+                return searchResult;
+            } catch (Exception e) {
+                return handleError(req, e);
+            }
+        });
+    }
+
+    private Object handleError(Request req, Exception e) {
+        log.error("Unexpected error in AiSearchApp: " + e.getMessage() + "\n\nrequest: " + req.body());
+        return "{}";
+    }
+
+    private String getUserKey() {
```

Comment:
I would suggest not to bring this method here not to violate single responsibility principle, rather it could be another method of RequestContext class